src/types.rs

```diff
@@ -336,7 +336,7 @@ impl Response {
     {
         Response::Other {
             code,
-            text: text.into(),
+            lines: vec![text.into()],
```

Comment:
This API could be "misused" when putting a string with a newline here. The imap-codec way to go about this would be to make a ...

```Rust
struct TextString { inner: String }
```

... that enforces the `1*(%d09 / %d32-126)` constraint on the `inner` value.

Still, your changes already make the code much better and I would want to let you decide if this is something you want to do right now or not. At some distant point in the future I would like to add structural tests for smtp-codec that check for such things, though.

Reply:
Okay, I've implemented the `TextString` type. I went with `Cow<'a, str>` for the inner value because I would like to avoid allocations and it seemed silly to work with `String` now only to rework it again soon.

Reply:
Awesome, thanks. I'll take another look tomorrow!